account/src/account/info.rs

```diff
@@ -0,0 +1,207 @@
+            let Ok(decoded) = serde_json::from_str::<GatewayAccount>(&text) else {
+                return Err(AccountError::CannotParseAccountInfo { address: address.to_string() })
+            };
+
+            let raw_info = decoded.data.account;
+            let Ok(balance) = BigUint::from_str(&raw_info.balance) else {
+                return Err(AccountError::CannotParseAccountBalance { address: bech32_address, balance: raw_info.balance})
+            };
+
+            let Ok(developer_reward) = BigUint::from_str(&raw_info.developer_reward) else {
+                return Err(AccountError::CannotParseAccountDeveloperReward { address: bech32_address, reward: raw_info.developer_reward})
+            };
+
+            let owner_address = if raw_info.owner_address.len() == 0 {
+                Address::default()
+            } else {
+                let Ok(address) = Address::from_bech32_string(&raw_info.owner_address) else {
+                    return Err(AccountError::CannotParseAccountOwnerAddress { address: bech32_address, owner: raw_info.owner_address})
+                };
+                address
+            };
+
+            let code_metadata = if let Some(raw_code_metadata) = raw_info.code_metadata {
+                Some(decode_code_meta_data(raw_code_metadata)?)
+            } else {
+                None
+            };
+
+            Ok(AccountInfo {
+                nonce: raw_info.nonce,
+                balance,
+                username: raw_info.username,
+                code: raw_info.code,
+                code_hash: raw_info.code_hash,
+                root_hash: raw_info.root_hash,
+                code_metadata: code_metadata,
+                developer_reward: developer_reward,
+                owner_address,
+            })
+        }
+    ).await
+}
+
+pub fn decode_code_meta_data(encoded: String) -> Result<CodeMetadata, AccountError> {
+    let decoded_bytes = base64::engine::general_purpose::STANDARD.decode(encoded.clone()).or(Err(AccountError::CannotDecodeCodeMetadata { metadata: encoded.clone() }))?;
+    if decoded_bytes.len() != 2 {
+        return Err(AccountError::CannotDecodeCodeMetadata { metadata: encoded });
+    }
+
+    let byte_array: [u8; 2] = decoded_bytes.as_slice().try_into().or(Err(AccountError::CannotDecodeCodeMetadata { metadata: encoded }))?;
+    Ok(CodeMetadata::from(byte_array))
+}
+
+#[cfg(test)]
+mod tests {
+    use num_bigint::BigUint;
+    use novax::caching::CachingNone;
+    use novax_data::Address;
+    use crate::account::info::fetch_account_info_for_address;
+    use crate::mock::request::MockClient;
+
+    #[tokio::test]
+    pub async fn test_with_valid_sc_address() {
+        let result = fetch_account_info_for_address(&MockClient::new(), &"erd1qqqqqqqqqqqqqpgqr7een4m5z44frr3k35yjdjcrfe6703cwdl3s3wkddz".into(), &CachingNone).await.unwrap();
+
+        assert_eq!(result.nonce, 0);
```

Comment:
Having one assert_eq! by field is error prone, because it's easy to miss a field. Can you derive PartialEq on AccountInfo and use one assert_eq! to compare the whole struct?